src/core/recv_buffer.c

```diff
@@ -186,6 +262,7 @@ QuicRecvBufferResize(
     _In_ uint32_t TargetBufferLength
     )
 {
+    CXPLAT_DBG_ASSERT(RecvBuffer->RecvMode != QUIC_RECV_BUF_MODE_EXTERNAL); // Should never resize in external mode
```

Comment:
FYI, we also have `CXPLAT_DBG_ASSERTMSG` which could be used here instead of the comment:
```suggestion
    CXPLAT_DBG_ASSERTMSG(RecvBuffer->RecvMode != QUIC_RECV_BUF_MODE_EXTERNAL, "Should never resize in external mode");
```
But you don't have to use it. We don't use it that much.